asmcomp/arm64/emit.mlp

```diff
@@ -339,33 +307,30 @@ let emit_cmpimm rs n =
   then `	cmp	{emit_reg rs}, #{emit_int n}\n`
   else `	cmn	{emit_reg rs}, #{emit_int (-n)}\n`
 
-(* Name of current function *)
-let function_name = ref ""
-(* Entry point for tail recursive calls *)
-let tailrec_entry_point = ref 0
-(* Pending floating-point literals *)
-let float_literals = ref ([] : (int64 * label) list)
-
 (* Label a floating-point literal *)
-let float_literal f =
+let float_literal env fl =
   try
-    List.assoc f !float_literals
+    let eq x =
+      Float.equal (Int64.float_of_bits x.fl) (Int64.float_of_bits fl)
```

Comment:
Sorry, I failed to notice that the floats were already stored as `int64`s, so the previous code (and the original `List.assoc` code) was correct. The whole point is to avoid using float comparison in the compiler, so this new code is wrong.
For clarity, I would still prefer `List.find (fun x -> Int64.equal x.fl fl) env.float_literals` rather than your original `compare`.
(If you follow this suggestion, please update `arm/emit.mlp` in the same way).

Reply:
Fixed in both arm and arm64.
Thank you for noticing this, it was my mistake (I misunderstood your comment to mean that Float comparison should be used instead of Int64).